.buildkite/scripts/imageprep.sh

```diff
@@ -2,23 +2,47 @@
 
 set -euo pipefail
 
-source .buildkite/scripts/setenv.sh
-
-DOCKER_REGISTRY = 'docker.elastic.co'
-DOCKER_REGISTRY_SECRET_PATH = "kv/ci-shared/platform-ingest/docker_registry_prod"
+DOCKER_REGISTRY_SECRET_PATH="kv/ci-shared/platform-ingest/docker_registry_prod"
 
 publish_docker_image() {
-    echo "Pushing the docker image "$DOCKER_IMAGE:$DOCKER_IMAGE_TAG" to the ${DOCKER_REGISTRY} registry..."
-    dockerLogin
-    docker push "$DOCKER_IMAGE:$DOCKER_IMAGE_TAG"
+    echo "Pushing the docker image "$DOCKER_IMAGE":"$DOCKER_IMAGE_TAG" to the "${DOCKER_REGISTRY}" registry..."
+    docker_login
+    docker push "${DOCKER_IMAGE}":"${DOCKER_IMAGE_TAG}"
     docker logout "$DOCKER_REGISTRY"
 }
 
-
-dockerLogin() {
-    DOCKER_USER=$(retry 5 vault kv get -field user ${DOCKER_REGISTRY_SECRET_PATH})
-    DOCKER_PASSWORD=$(retry 5 vault kv get -field password ${DOCKER_REGISTRY_SECRET_PATH})
+docker_login() {
+    DOCKER_USER=$(retry 5 vault kv get -field user "${DOCKER_REGISTRY_SECRET_PATH}")
+    DOCKER_PASSWORD=$(retry 5 vault kv get -field password "${DOCKER_REGISTRY_SECRET_PATH}")
     docker login -u "${DOCKER_USER}" -p "${DOCKER_PASSWORD}" "${DOCKER_REGISTRY}" 2>/dev/null
+    trap cleanup EXIT
```

Comment:
If more cleanup processes are added later, this call could be moved to the top of this shell script. But, there is not need to move it now.